proto/internal/temporal/server/api/persistence/v1/chasm.proto

```diff
@@ -61,11 +61,24 @@ message ChasmComponentAttributes {
         temporal.api.common.v1.DataBlob data = 4;
         // Versioned transition of the Entity when the task was created.
         VersionedTransition versioned_transition = 5;
+        // The xth task generated in this versioned transition.
+        // Together with the versioned transition, this is a unique identifier for
+        // this task.
+        int64 versioned_transition_offset = 6;
+        // If a physical task is created for this task in this cluster.
+        // NOTE: this is a cluster-specific field and can not be replicated.
+        // Changes to this field also doesn't require an increase in versioned transition. 
+        int32 physical_task_status = 7;
     }
 
     // Fully qualified type name of a registered component.
     string type = 1;
-    repeated Task tasks = 3;
+    // Tasks are in their insertion order, 
+    // i.e. by versioned transtion and index_in_transition.
+    repeated Task side_effect_tasks = 2;
+    // Tasks are ordered by their scheduled time, breaking ties by
+    // versioned transition and index_in_transition. 
+    repeated Task pure_tasks = 3;
```

Comment:
Separated those two tasks type so we can order them differently and make it more efficient for pure task processing. The processing logic doesn't have to loop through all tasks and see if the task is a pure task and is due for processing.